lib/withOnyx.js

```diff
@@ -60,8 +61,10 @@ export default function (mapOnyxToState, shouldDelayUpdates = false) {
                     (resultObj, mapping, propertyName) => {
                         const key = Str.result(mapping.key, props);
                         let value = OnyxUtils.tryGetCachedValue(key, mapping);
-                        if (!value && mapping.initialValue) {
-                            value = mapping.initialValue;
+                        const hasCacheForKey = cache.hasCacheForKey(key);
+
+                        if (!hasCacheForKey && !value && mapping.initialValue) {
+                            value = mapping.initialValue === null ? undefined : mapping.initialValue;
```

Comment:
`null` is a falsy value in javascript so how could we get here with `mapping.initialValue === null` if we check for truthy `mapping.initialValue` above 🤔

Reply:
yes, you're right, This doesn't make sense